am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -203,6 +213,11 @@ private boolean explicitAccessType(Set<String> userRoles) {
         return roleBasedAccessRecords.stream().collect(getMapCollector());
     }
 
+    private boolean explicitAccessType(Set<String> userRoles) {
```

Comment:
Would be better to have the following order:

* public filter list of resources
* public filter resource
* private method that check whether role has explicit access

This way code reads like a book.

Reply:
Nice, that made nice clean diff